src/domain/error.rs

```diff
@@ -0,0 +1,8 @@
+pub type LoadBalancerResult<T> = Result<T, LoadBalancerError>;
+
+#[derive(Debug)]
```

Comment:
I'd suggest considering usage of `thiserror` crate or similar.

Reply:
Yes, good point. I've created an issue to address this: https://github.com/thedrummeraki/capstone-rust/issues/21